### PR TITLE
Rx Refactor SuggestionsViewModel

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
Rx Refactor SuggestionsViewModel

### Relationships
Part of #3352

## Why do we want this?
To have reactive code.

## How is it done?
By getting rid of Mvx binding and replace it with Rx.

### Why not another way?
One could use `ObservableGroupedOrderedCollection`. However, since the suggestion list is immutable, adopting that class is kinda overkill. so I'm going for `IObservable<Suggestion[]>`

### Side effects
None

## Review hints
Don't go commit by commit, review the whole thing at once. Follow the VM -> VC (Activity) flow

## :squid: Permissions
🐙 after getting one approval from each platform.

### CODE REVIEW

**Comment by khoi** [comment]
WIP as the test is failing

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/BaseViewModelTests.cs [hunk 1]
```diff
@@ -34,7 +36,9 @@ public abstract class BaseViewModelTests<TViewModel> : BaseMvvmCrossTests
         protected IRemoteConfigService RemoteConfigService { get; } = Substitute.For<IRemoteConfigService>();
         protected IPasswordManagerService PasswordManagerService { get; } = Substitute.For<IPasswordManagerService>();
         protected IErrorHandlingService ErrorHandlingService { get; } = Substitute.For<IErrorHandlingService>();
-        protected ISuggestionProviderContainer SuggestionProviderContainer { get; } = Substitute.For<ISuggestionProviderContainer>();
+
+        protected ISuggestionProviderContainer SuggestionProviderContainer =
```
Comment (sofokli1) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/BaseViewModelTests.cs]: why remove the`{ get; }` only for this one?

File: Toggl.Daneel/Views/Suggestions/SuggestionsView.cs [hunk 2]
```diff
@@ -118,14 +69,39 @@ private void prepareSuggestionViews()
...
                + index * distanceBetweenSuggestions
                + index * suggestionHeight;
+
+        private float heightForSuggestionCount(int count)
+        {
+            if (count == 0)
+            {
+                return 0;
+            }
+            return count * (suggestionHeight + distanceBetweenSuggestions) + distanceAbowTitleLabel
```
[the remark this answers is not in the document]
Comment (khoi) [reply Toggl.Daneel/Views/Suggestions/SuggestionsView.cs]: I copied this calculation from the previous implementation. I believe the layout is currently like this. 

For instance: count = 2

`suggestionHeight` | `distanceBetween` | `suggestionHeight` | `distanceBetween`

if we do `count - 1` * `distanceBetween` the shadow would be cut off

screen shot 2018-10-25 at 18 30 05

File: Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs [hunk 3]
```diff
@@ -16,92 +19,63 @@ namespace Toggl.Foundation.MvvmCross.ViewModels
...
-        {
-            Suggestions.Add(suggestion);
-
-            RaisePropertyChanged();
+                .ToArray()
+                .AsDriver(onErrorJustReturn: new Suggestion[0], schedulerProvider: schedulerProvider);
+
+            IsEmpty = Suggestions
+                .Select(suggestions => suggestions.Length)
+                .Select(c => c == 0)
```
Comment (Odrakir) [inline Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs]: I think this is better as just one `Select`
```suggestion
                .Select(suggestions => suggestions.Length == 0)
```

**Pull request merged**